step-controller-plugins/step-controller-plugins-timeseries/src/main/java/step/plugins/timeseries/TimeSeriesService.java

```diff
@@ -28,28 +26,35 @@
 import java.util.Collections;
 import java.util.Map;
 import java.util.Objects;
+import java.util.Set;
 import java.util.concurrent.atomic.LongAdder;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
+import static step.plugins.timeseries.TimeSeriesControllerPlugin.RESOLUTION_PERIOD_PROPERTY;
+import static step.plugins.timeseries.TimeSeriesControllerPlugin.TIME_SERIES_COLLECTION_PROPERTY;
+
 @Singleton
 @Path("/time-series")
 @Tag(name = "TimeSeries")
 public class TimeSeriesService extends AbstractStepServices {
 
     private AsyncTaskManager asyncTaskManager;
     private TimeSeriesAggregationPipeline aggregationPipeline;
-    private TimeSeriesIngestionPipeline ingestionPipeline;
     private Collection<Measurement> measurementCollection;
+    private TimeSeries timeSeries;
+    private Integer ingestionResolutionPeriod;
 
     @PostConstruct
     public void init() throws Exception {
         super.init();
         GlobalContext context = getContext();
         aggregationPipeline = context.require(TimeSeriesAggregationPipeline.class);
-        ingestionPipeline = context.require(TimeSeriesIngestionPipeline.class);
         asyncTaskManager = context.require(AsyncTaskManager.class);
         measurementCollection = context.getCollectionFactory().getCollection(MeasurementAccessor.ENTITY_NAME, Measurement.class);
+        CollectionFactory collectionFactory = context.getCollectionFactory();
+        timeSeries = new TimeSeries(collectionFactory, TIME_SERIES_COLLECTION_PROPERTY, Set.of());
```

Comment:
@lucianbaciu Requiring the TimeSeries instance from the context would be a better option than recreating an instance here. It avoid the need of exposing all the instanciation parameters (entityName, resolution, etc).

Reply:
agree